From c2c9faf5bb57baae3a595d460bbfcdcdd28ff3ed Mon Sep 17 00:00:00 2001
From: Pascal Paillet <p.paillet@foss.st.com>
Date: Thu, 27 Jun 2024 18:41:22 +0200
Subject: [PATCH 625/954] feat(stm32mp21): describe power supplies on
 STM32MP215F-DK board

Describe the power supplies of stm32mp215f-dk board.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 51082ceb81de42f370eefe580f47549a1b5b4e12 feat(stm32mp21): describe power supplies on STM32MP215F-DK board]
Change-Id: I048f1a9d09a740683727adad1d0a25edf4eb9050
Signed-off-by: Pascal Paillet <p.paillet@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/390254
Tested-by: Pascal PAILLET <p.paillet@st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Pascal PAILLET <p.paillet@st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
---
 fdts/stm32mp215f-dk.dts | 88 +++++++++++++++++++++++++++++++++++++++++
 1 file changed, 88 insertions(+)

diff --git a/fdts/stm32mp215f-dk.dts b/fdts/stm32mp215f-dk.dts
index 344338d477..c6b205eee0 100644
--- a/fdts/stm32mp215f-dk.dts
+++ b/fdts/stm32mp215f-dk.dts
@@ -43,6 +43,76 @@
 	status = "okay";
 	#address-cells = <1>;
 	#size-cells = <0>;
+
+	pmic2: pmic@33 {
+		compatible = "st,stpmic2l";
+		reg = <0x33>;
+		status = "okay";
+
+		regulators {
+			compatible = "st,stpmic2-regulators";
+
+			vddcore: buck1 {
+				regulator-name = "vddcore";
+				regulator-min-microvolt = <820000>;
+				regulator-max-microvolt = <820000>;
+				regulator-always-on;
+			};
+			vdd2_ddr: buck2 {
+				regulator-name = "vdd2_ddr";
+				regulator-min-microvolt = <1100000>;
+				regulator-max-microvolt = <1100000>;
+			};
+			vddcpu: buck3 {
+				regulator-name = "vddcpu";
+				regulator-min-microvolt = <800000>;
+				regulator-max-microvolt = <910000>;
+				regulator-always-on;
+			};
+			vdda1v8_aon: ldo1 {
+				regulator-name = "vdda1v8_aon";
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <1800000>;
+				regulator-always-on;
+				st,mask-reset;
+			};
+			vddio_pmic: ldo2 {
+				regulator-name = "vddio_pmic";
+				regulator-min-microvolt = <3300000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-always-on;
+				st,mask-reset;
+			};
+			vdd1_ddr: ldo3 {
+				regulator-name = "vdd1_ddr";
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <1800000>;
+				regulator-enable-ramp-delay = <1000>;
+			};
+			vdd3v3_usb: ldo4 {
+				regulator-name = "vdd3v3_usb";
+				regulator-min-microvolt = <3300000>;
+				regulator-max-microvolt = <3300000>;
+			};
+			vdd_flash: ldo5 {
+				regulator-name = "vdd_flash";
+				regulator-min-microvolt = <3300000>;
+				regulator-max-microvolt = <3300000>;
+			};
+			vdda_1v8: ldo6 {
+				regulator-name = "vdda_1v8";
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <1800000>;
+				regulator-always-on;
+			};
+			v3v3: gpo1 {
+				regulator-name = "v3v3";
+				regulator-min-microvolt = <3300000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-always-on;
+			};
+		};
+	};
 };
 
 &iwdg1 {
@@ -54,6 +124,24 @@
 	status = "okay";
 };
 
+&pwr {
+	vddio1 {
+		vddio1-supply = <&vddio_pmic>;
+	};
+
+	vddio2 {
+		vddio2-supply = <&vdda_1v8>;
+	};
+
+	vddio3 {
+		vddio3-supply = <&vddio_pmic>;
+	};
+
+	vddio {
+		vdd-supply = <&vddio_pmic>;
+	};
+};
+
 &rng1 {
 	status = "okay";
 };
-- 
2.53.0

